src/components/Organization/Team.tsx

```diff
@@ -94,6 +145,356 @@ export const useAllTeamMembers = () => {
+
+  // This function is not working right now.
+  const resendInvitation = () => {
+    inviteMember.mutate(member, {
+      onSuccess: () => {
+        toast({
+          title: t('team.actions.resend_invitation_success', { defaultValue: 'Invitation resent successfully' }),
+          status: 'success',
+          duration: 5000,
+          isClosable: true,
+        })
+      },
+      onError: (error: Error) => {
+        toast({
+          title: t('team.actions.resend_invitation_error', { defaultValue: 'Error resending invitation' }),
+          description: error.message,
+          status: 'error',
+          duration: 5000,
+          isClosable: true,
+        })
+      },
+    })
+  }
+
+  return (
+    <Button leftIcon={<Icon as={LuRefreshCw} />} fontSize='sm' variant='ghost' onClick={resendInvitation}>
+      {t('team.actions.resend_invitation', { defaultValue: 'Resend Invitation' })}
+    </Button>
+  )
+}
+
+const ActiveMemberActions = ({ onOpenRole, onOpenRemove }) => {
+  const { t } = useTranslation()
+
+  return (
+    <>
+      <Button onClick={onOpenRole} leftIcon={<Icon as={LuUserCog} />} fontSize='sm' variant='ghost'>
+        {t('team.actions.change_role', { defaultValue: 'Change Role' })}
+      </Button>
+      <PopoverFooter>
+        <Button variant='ghost' colorScheme='red' fontSize='sm' onClick={onOpenRemove}>
+          {t('team.actions.remove_member', { defaultValue: 'Remove Member' })}
+        </Button>
+      </PopoverFooter>
+    </>
+  )
+}
+
+const MemberActions = ({ member }) => {
+  const { t } = useTranslation()
+  const { isOpen: isRoleModalOpen, onOpen: openRoleModal, onClose: closeRoleModal } = useDisclosure()
+  const { isOpen: isRemoveModalOpen, onOpen: openRemoveModal, onClose: closeRemoveModal } = useDisclosure()
+  const { data: profile, isLoading } = useProfile()
+  const hasId = Boolean(member.info?.id)
+  const isCurrentUser = member?.info.id === profile?.id
+
+  if (isLoading) return <Progress isIndeterminate />
+
+  if (isCurrentUser) return null
+
+  return (
+    <>
+      <Popover placement='bottom-end' isLazy>
+        <PopoverTrigger>
+          <IconButton icon={<Icon as={LuEllipsis} />} ml='auto' variant='transparent' aria-label='Options' />
+        </PopoverTrigger>
+        <PopoverContent w='auto'>
+          <Text fontWeight='extrabold' p={2} fontSize='sm'>
+            {t('team.actions.title', { defaultValue: 'Actions' })}
+          </Text>
+          {hasId ? (
+            <ActiveMemberActions onOpenRole={openRoleModal} onOpenRemove={openRemoveModal} />
+          ) : (
+            <PendingInvitationActions member={member} />
+          )}
+        </PopoverContent>
+      </Popover>
+      {hasId && (
+        <>
+          <ChangeRoleModal isOpen={isRoleModalOpen} onClose={closeRoleModal} member={member} />
+          <RemoveMemberModal isOpen={isRemoveModalOpen} onClose={closeRemoveModal} member={member} />
+        </>
+      )}
+    </>
+  )
+}
```

Comment:
These two components should be using the Menu component instead of popover. Also, use the isLoading from the Menu Button rather than returning a progress.